client/src/pages/PostSolution.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { useState, ChangeEvent, FormEvent } from "react";
+import { useParams } from "react-router-dom";
+import { Col, Container, Form, Row } from "react-bootstrap";
+import Button from "../components/Button.tsx";
+
+interface FormData {
+  title: string;
+  description: string;
+  diagram: File | null;
+}
+
+// TODO: Add form validation
+// TODO: Include data about the challenge like title
+
+const PostSolution: React.FC = () => {
+  const { challengeId } = useParams();
+  const [formData, setFormData] = useState<FormData>({
+    title: "",
+    description: "",
+    diagram: null,
+  });
+
+  const handleChange = (
+    e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({
+      ...prevData,
+      [name]: value,
+    }));
+  };
+
+  const handleFileChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (e.target.files && e.target.files.length > 0) {
+      const file = e.target.files[0];
+      setFormData((prevData) => ({
+        ...prevData,
+        diagram: file,
+      }));
+    }
+  };
+
+  const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // Handle form submission here
+    console.log(formData);
+    console.log(challengeId);
+  };
+  return (
+    <Container>
+      <Row className={"my-5 justify-content-center"}>
+        <Col md={4} className={"bg-gray-200 rounded py-5 px-md-5"}>
+          <h2>Submit Your Solution!</h2>
```

Comment:
Maybe change it to `<h1 className="fw-2">`

Its a good practice to always have an h1 in your page.

Reply:
done